src/components/OrdersContainer/Orders/OrderForm/useFormFields.ts

```diff
@@ -0,0 +1,97 @@
+import { OrderFieldDescription, ValueDescription } from "adapters/Order";
+import { useState, useEffect } from "react";
+
+export enum FieldState {
+    DEFAULT,
+    VALID,
+    INVALID,
+}
+
+export type FormField = {
+    name: string;
+    valueDescription: ValueDescription;
+    value: string | boolean | number;
+    isValid: boolean;
+    isEnabled: boolean;
+};
+
+function getInitialFormFields(
```

Comment:
this could be splitted into multiple functions

```suggestion
function getInitialFormFields(
    orderFields: Record<string, OrderFieldDescriptions>
): FormField[] {
    return Object.entries(orderFields).map(([_, fieldDescription]) => {
        return getFormField(fieldDescription)
    }
}

function getFormField(fieldDescription: OrderFieldDescription): FormField {
    if (fieldDescription.valueDescription.kind == "numeric") {
        return getNumericFormField(fieldDescription)
    } else if (fieldDescription.valueDescription.kind == "boolean") {
        return getBooleanFormField(fieldDescription)
    } else {
        return getGenericFormField(fieldDescription)
    }
}
        
function getNumericFormField(fieldDescription: OrderFieldDescription): FormField {
    return {
        name: fieldDescription.name,
        valueDescripion: fieldDescription.valueDescription,
        value: 0,
        isValid: false,
        isEnabled: true,
    }
}

function getBooleanFormField(fieldDescription: OrderFieldDescription): FormField {
    return {
        name: fieldDescription.name,
        valueDescripion: fieldDescription.valueDescription,
        value: false,
        isValid: true,
        isEnabled: true,
    }
}

function getGenericFormField(fieldDescription: OrderFieldDescription): FormField {
    return {
        name: fieldDescription.name,
        valueDescripion: fieldDescription.valueDescription,
        value: fieldDescription.valueDescription.value[0],
        isValid: true,
        isEnabled: true,
    }
}
```